packages/zosfiles/__tests__/cli/mount/fs/fs.handler.unit.test.ts

```diff
@@ -0,0 +1,180 @@
+
+            // Vars populated by the mocked function
+            let error;
+            let apiMessage = "";
+            let jsonObj;
+            let logMessage = "";
+            let fakeSession = null;
+
+            // Mock the zfs function
+            Mount.fs = jest.fn((session) => {
+                fakeSession = session;
+                return {
+                    success: true,
+                    commandResponse: "mounted"
+                };
+            });
+
+            // Mocked function references
+            const profFunc = jest.fn((args) => {
+                return {
+                    host: "fake",
+                    port: "fake",
+                    user: "fake",
+                    password: "fake",
+                    auth: "fake",
+                    rejectUnauthorized: "fake",
+                };
+            });
+
+            try {
+                // Invoke the handler with a full set of mocked arguments and response functions
+                await handler.process({
+                    arguments: {
+                        $0: "fake",
+                        _: ["fake"],
+                        fileSystemName,
+                        mountPoint,
+                        ...UNIT_TEST_ZOSMF_PROF_OPTS
+                    },
+                    response: {
+                        data: {
+                            setMessage: jest.fn((setMsgArgs) => {
+                                apiMessage = setMsgArgs;
+                            }),
+                            setObj: jest.fn((setObjArgs) => {
+                                jsonObj = setObjArgs;
+                            })
+                        },
+                        console: {
+                            log: jest.fn((logArgs) => {
+                                logMessage += "\n" + logArgs;
+                            })
+                        },
+                        progress: {
+                            startBar: jest.fn((parms) => {
+                                // do nothing
+                            }),
+                            endBar: jest.fn(() => {
+                                // do nothing
+                            })
+                        }
+                    },
+                    profiles: {
+                        get: profFunc
+                    }
+                } as any);
+            } catch (e) {
+                error = e;
+            }
+
+            expect(error).toBeUndefined();
+            expect(profFunc).toHaveBeenCalledWith("zosmf", false);
+            expect(Mount.fs).toHaveBeenCalledTimes(1);
+            expect(Mount.fs).toHaveBeenCalledWith(fakeSession, fileSystemName, mountPoint, {});
+            expect(jsonObj).toMatchSnapshot();
+            expect(apiMessage).toMatchSnapshot();
+            expect(logMessage).toMatchSnapshot();
+        });
+    });
+
+    it("should raise an error", async () => {
```

Comment:
Can you make this title a bit more specific? Maybe include what error?

Reply:
Not sure if we can be more specific about the type of error, its just a fake one that we mocked out in the handler.

Reply:
It's not a big deal... sorry, I should have been more clear... What exactly is this test validating? The describe/its are supposed to read like a sentence: "Mount file system handler should raise an error" ?? Why? Does it always raise an error?

 The other test you have is more clear "Mount file system handler should mount a FS if requested".

Reply:
This test was copied over from other unit test handler suites that had the same one. Renamed it to "should fail to mount a FS when there is an error" which is hopefully a bit clearer. I believe the intent is to verify that if the `Mount.fs` API function fails, the handler function for the `mount` command should reflect that by returning an error.

Reply:
Your explanation of the intention of the test is what I was originally looking for... I know that we're not consistent everywhere. I think some of the tests have verbiage more like... "should not transform an error thrown from the rest client" or something along those lines.